sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1233,26 +1232,19 @@ def _retry_region_zones(self,
             # down the non-existent cluster will itself error out with the same
             # error message.  This was found to be confusing. In that case we
             # skip termination.
+            skip_cleanup = not cluster_exists and definitely_no_nodes_launched
+            if skip_cleanup:
+                continue
```

Comment:
Tried both and the failover proceeded normally in both cases. This is because even if we don't remove the cluster record/generated yaml/SSH config, when we try the next region/zone these information will be overwritten.  Wdyt?

Reply:
That makes sense, but the comment above the if statement seems staled now, as the `teardown_no_lock` works for the cluster even if the error occurs. Should we update it?

Reply:
Do you mean
```
            # If these conditions hold, it *should* be safe to skip the cleanup
            # action.
            #
            # We want to skip mainly for custom VPC: if users encountered "No
            # VPC with name 'xxx' is found in <region>.", then going ahead to
            # down the non-existent cluster will itself error out with the same
            # error message.  This was found to be confusing. In that case we
            # skip termination.
            skip_cleanup = not cluster_exists and definitely_no_nodes_launched
            if skip_cleanup:
                continue
```
? I think this comment is still accurate. If we proceed to do `CloudVmRayBackend().teardown_no_lock()`, another error `SKYPILOT_ERROR_NO_NODES_LAUNCHED` will be raised during the teardown. It's only safe to proceed to do `post_teardown_cleanup`.